jol-core/src/test/java/org/openjdk/jol/vm/HotspotUnsafeTest.java

```diff
@@ -0,0 +1,25 @@
+package org.openjdk.jol.vm;
+
+import org.junit.Test;
+
+import java.lang.management.ManagementFactory;
+import java.util.Optional;
+
+import static org.apache.commons.lang3.StringUtils.substringAfter;
+import static org.junit.Assert.assertEquals;
+
+public class HotspotUnsafeTest {
+    @Test
+    public void testAlignment(){
+        assertEquals(expectedAlignment(), VM.current().objectAlignment());
+    }
+
+    private int expectedAlignment() {
+        Optional<String> alignmentProperty = ManagementFactory.getRuntimeMXBean().getInputArguments().stream()
+                .filter(arg -> arg.contains("-XX:ObjectAlignmentInBytes"))
+                .findAny();
+        return alignmentProperty
+                .map(alignment -> Integer.parseInt(substringAfter(alignment, "=")))
+                .orElseGet(() -> Integer.parseInt(System.getProperty("sun.arch.data.model")) / Byte.SIZE);
```

Comment:
This is not really correct. On 32-bit, the object alignment is not `32/8 = 4`, it is still `8`. Hotspot just does `8` on these platforms, put `8` in here too.